scripts/us_census/acs5yr/subject_tables/common/generate_col_map.py

```diff
@@ -0,0 +1,410 @@
+        stat_var = {}
+        # part_list contains a list of tokens of the column name after splitting the
+        # column name based on the '!!' delimiter
+        part_list = column.split(self.delimiter)
+
+        ## set the measurement for special cases: Median, Mean, etc.
+        ## Pass 1: Check if the entire column is present in spec
+        if not measurement_assigned and column in self.features['measurement']:
+            stat_var.update(self.features['measurement'][column])
+            measurement_assigned = True
+
+        ## Pass 2: Check if a token of the column is present in spec
+        for part in part_list:
+            # set the base for special cases like median, etc.
+            if not measurement_assigned and part in self.features['measurement']:
+                stat_var.update(self.features['measurement'][part])
+                measurement_assigned = True
+
+        # set the default statVar definition
+        if not measurement_assigned and '_DEFAULT' in self.features[
+                'measurement']:
+            stat_var.update(self.features['measurement']['_DEFAULT'])
+
+        # set the populationType attribute
+        if 'populationType' not in stat_var:
+            stat_var['populationType'] = self._get_population_type(part_list)
+
+        # TODO: To check if there are edge-cases with values not being full columns
+        # associate pvs to stat_var
+        for part in part_list:
+            ##  p = property and k = dictionary of substrings in column that points to a propertyValue
+            for p, k in self.features['pvs'].items():
+                ## check if the current key of dict matches with any substring from the tokens of a column name
+                for c, v in k.items():
+                    # check if column substring matches with a column token
+                    # the check is done by ignoring the case
+                    if part.lower() == c.lower():
+                        if p in stat_var:
+                            logger.warning(
+                                f"For column: {column} | Property {p} has an existing value {stat_var[p]} which is modified to value {p}"
+                            )
+                        stat_var[p] = v
+
+        ## add Universe PVs based on the populationType of StatVar
+        dependent_properties = None
+        for elem in self.features['universePVs']:
+            if stat_var['populationType'] == elem['populationType']:
+                # check if all constraints of this populationType is in stat_var
+                if (set(elem['constraintProperties']).issubset(
+                        set(list(stat_var.keys())))):
+                    try:
+                        ## if the dependentPVs are not in statVar add them
+                        if not set(list(elem['dependentPVs'].keys())).issubset(
+                                set(list(stat_var.keys()))):
+                            stat_var.update(elem['dependentPVs'])
```

Comment:
safer to set values only for properties not already in stat_var so as to not overwrite an existing property.
can be addressed in a future TOD as the current tables use an exclusive set of dependent PVs.